sdk/schemaregistry/azure-data-schemaregistry/src/test/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClientTests.java

```diff
@@ -0,0 +1,307 @@
+package com.azure.data.schemaregistry;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.test.TestBase;
+import com.azure.data.schemaregistry.implementation.models.ServiceErrorResponseException;
+import com.azure.data.schemaregistry.models.SerializationType;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.nio.charset.StandardCharsets;
+import java.time.OffsetDateTime;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Pattern;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SchemaRegistryAsyncClientTests extends TestBase {
+    private static final int RESOURCE_LENGTH = 16;
+    private static final String AZURE_EVENTHUBS_FULLY_QUALIFIED_DOMAIN_NAME = "AZURE_EVENTHUBS_FULLY_QUALIFIED_DOMAIN_NAME";
+    private static final String SCHEMA_CONTENT = "{\"type\" : \"record\",\"namespace\" : \"TestSchema\",\"name\" : \"Employee\",\"fields\" : [{ \"name\" : \"Name\" , \"type\" : \"string\" },{ \"name\" : \"Age\", \"type\" : \"int\" }]}";
+    private static final String SCHEMA_GROUP = "at";
+    private static final Pattern WHITESPACE_PATTERN = Pattern.compile("\\s+", Pattern.MULTILINE);
+    private static final String SCHEMA_CONTENT_NO_WHITESPACE = WHITESPACE_PATTERN.matcher(SCHEMA_CONTENT).replaceAll("");
+
+    SchemaRegistryClientBuilder builder;
+
+    @Override
+    protected void afterTest() {
+        Mockito.framework().clearInlineMocks();
+        super.afterTest();
+    }
+
+    /**
+     * Verifies that we can register a schema and then get it by its schemaId.
+     */
+    @Test
+    public void registerAndGetSchema() {
+        // Arrange
+        initializeBuilder();
```

Comment:
You can override `beforeTest()` method defined in `TestBase` to initialize the builder instead of calling this from each test method.

Reply:
It used to be a parameterised test that would take a different HttpClient because I thought that's what we wanted to test.

https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTest.java#L91